test/e2e-wi-tests.sh

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname $0)/e2e-common.sh
+
+# Override the setup and teardown functions to install wi-enabled control plane components.
+readonly K8S_CONTROLLER_SERVICE_ACCOUNT="controller"
+readonly PROW_SERVICE_ACCOUNT=$(gcloud config get-value core/account)
+
+if (( ! IS_PROW )); then
+  readonly AUTHENTICATED_SERVICE_ACCOUNT="${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com"
+  readonly MEMBER="serviceAccount:${E2E_PROJECT_ID}.svc.id.goog[${CONTROL_PLANE_NAMESPACE}/${K8S_CONTROLLER_SERVICE_ACCOUNT}]"
+  readonly DATA_PLANE_SERVICE_ACCOUNT="${PUBSUB_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com"
+else
+  readonly AUTHENTICATED_SERVICE_ACCOUNT=${PROW_SERVICE_ACCOUNT}
+  readonly MEMBER="serviceAccount:${PROJECT}.svc.id.goog[${CONTROL_PLANE_NAMESPACE}/${K8S_CONTROLLER_SERVICE_ACCOUNT}]"
+  # Get the PROW service account.
+  readonly PROW_PROJECT_NAME=$(cut -d'.' -f1 <<< $(cut -d'@' -f2 <<< ${AUTHENTICATED_SERVICE_ACCOUNT}))
+  readonly DATA_PLANE_SERVICE_ACCOUNT=${AUTHENTICATED_SERVICE_ACCOUNT}
+fi
+
+# Create resources required for the Control Plane setup.
+function knative_setup() {
+  control_plane_setup || return 1
+  start_knative_gcp "workloadIdentityEnabled" || return 1
+  kubectl annotate serviceaccount ${K8S_CONTROLLER_SERVICE_ACCOUNT} iam.gke.io/gcp-service-account=${AUTHENTICATED_SERVICE_ACCOUNT} \
+    --namespace ${CONTROL_PLANE_NAMESPACE}
+}
+
+function control_plane_setup() {
```

Comment:
This file seems very similar to `test/e2e-common.sh`, is there any chance you can reuse(or refactor and reuse) some logic there such as `control_plane_setup`?

Reply:
e2e-common.sh is for non-wi script. We will finally use the init script to replace those gcloud command with the issue #738. After that, they will not be similar. The similarity are all for the gcloud command to grant permission in non-PROW environment. Keep it as it is for now.